scripting/get5/teamlogic.sp

```diff
@@ -18,22 +18,16 @@ public Action Timer_PlacePlayerOnJoin(Handle timer, int userId) {
 
 public void CheckClientTeam(int client) {
   Get5Team correctTeam = GetClientMatchTeam(client);
-  char auth[AUTH_LENGTH];
   int csTeam = Get5TeamToCSTeam(correctTeam);
   int currentTeam = GetClientTeam(client);
 
   if (csTeam != currentTeam) {
-    if (IsClientCoaching(client)) {
-      UpdateCoachTarget(client, csTeam);
-    } else if (GetAuth(client, auth, sizeof(auth))) {
-      char steam64[AUTH_LENGTH];
-      ConvertAuthToSteam64(auth, steam64);
-      if (IsAuthOnTeamCoach(steam64, correctTeam)) {
-        UpdateCoachTarget(client, csTeam);
-      }
+    if (!CheckIfClientCoachingAndMoveToCoach(client, correctTeam)) {
```

Comment:
Since you run `CheckIfClientCoachingAndMoveToCoach` in any case, this whole block can be changed to:

```c
if (!CheckIfClientCoachingAndMoveToCoach(client, correctTeam)) {
  if (csTeam != currentTeam) {
    SwitchPlayerTeam(client, csTeam);
  }
}
```